js/apps/admin-ui/src/identity-providers/add/DetailSettings.tsx

```diff
@@ -114,6 +160,40 @@ const Header = ({ onChange, value, save, toggleDeleteDialog }: HeaderProps) => {
         )}
         divider={false}
         dropdownItems={[
+          ...(provider?.providerId?.includes("saml") &&
+          getValues("config.validateSignature") === "true" &&
+          getValues("config.useEntityDescriptorUrl") === "true" &&
+          getValues("config.entityDescriptorUrl") &&
```

Comment:
It might be better to watch these values, as `getValues()` will not re-trigger if the value changes.

Reply:
The actions are only shown if `!formState.isDirty` because I don't want the user can do weird things with certificates if they are changing the provider configuration. So the actions can only be clicked if the configuration is saved. Do I need the watches in that case for both comments? (Sorry but with react I usually just copy&paste you are doing, :smile: but I didn't want to bother you with this until we decide to include it.)

Reply:
Yeah, I would recommend to add the watches for both, but they can be the same variable. You can do so by adding the following code:

```ts
const { setValue, formState, control } = useFormContext();

const validateSignature = useWatch({
  control,
  name: "config.validateSignature",
});

const useEntityDescriptorUrl = useWatch({
  control,
  name: "config.useEntityDescriptorUrl",
});

const entityDescriptorUrl = useWatch({
  control,
  name: "config.entityDescriptorUrl",
});
```

Reply:
OK then, it's done now.